src/rdstring.c

```diff
@@ -632,6 +761,68 @@ static int ut_string_split(void) {
         RD_UT_PASS();
 }
 
+/**
+ * @brief Unittest for rd_string_split_csv()
+ */
+static int ut_string_split_csv(void) {
+        static const struct {
+                const char *input;
+                const char sep;
+                rd_bool_t skip_empty;
+                size_t exp_cnt;
+                const char *exp[4];
+        } tests[] = {
+            {
+                "extension1:extension1val,\"extension2:extension2val1,"
+                "extension2val2\"",
+                ',',
+                rd_true,
+                2,
+                {"extension1:extension1val",
+                 "extension2:extension2val1,extension2val2"},
+            },
+            {
+                "extension1:extension1val,extension2:extension2val1",
+                ',',
+                rd_true,
+                2,
+                {"extension1:extension1val", "extension2:extension2val1"},
+            },
```

Comment:
Can we also add other edge cases like empty string, string with just a single k:v pair, and using escape sequences. And also test for invalid strings.